### PR TITLE
Fix: Expand as we traverse lineage

### PR DESCRIPTION
**Pull request by vchan**
This PR
- Recursively expands expressions and extract source aliases as we traverse lineage
- Removes cache of source tables because we were losing columns when a query references multiple columns from the same source table

### CODE REVIEW

File: sqlglot/lineage.py [hunk 1]
```diff
@@ -110,18 +108,17 @@ def to_node(
         for c in set(select.find_all(exp.Column)):
             table = c.table
             source = scope.sources[table]
+            if source.expression and sources:
+                expression = exp.expand(source.expression, parsed_sources)
```
Comment (tobymao) [inline sqlglot/lineage.py:112]: why is this needed? why don't you do this in the pre?

Comment (vchan) [reply sqlglot/lineage.py:112]: Okay, I'll update expand.

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -5059,7 +5064,7 @@ def _expand(node: Expression):
             if source:
                 subquery = source.subquery(node.alias or name)
                 subquery.comments = [f"source: {name}"]
-                return subquery
+                return subquery.transform(_expand, copy=copy)
```
Comment (tobymao) [inline sqlglot/expressions.py:5067]: i think copy can always be False here because teh parent respects the copy
